transcom/mymove
# E 04823 tickets b 18472 and b 18457 int

## [Feature E-04823](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A866910)
- [Frontend: B-18472](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A887939)
- [Backend: B-18457](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A886626)

## Summary

These two tickets introduce new business logic and refactoring with how service member duty location, grade, and weight allotments function within MilMove. At a high level, grade and duty location has moved from the service member to the order. Additionally, grades now reflect a "true" value that is calculated by weight allotment and assigned to a `DBWeightAuthorized`. `DBWeightAuthorized` is directly tied to the maximum billable weight of a move. When a service member grade is updated, it immediately updated the maximum billable weight, whether it be increasing or decreasing the maximum amount. 

## Verification Steps for Reviewers


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as a TIO
2. Select a move
3. Navigate to move details -> allowances
4. Make note of the "Weight allowance"
5. Update the grade to something that will change the weight allowance (See `pkg/models/entitlements.go`)
6. Review
7. Login as a Counselor
8. Select a move
9. Navigate to move details -> allowances
10. Make note of the "Weight allowance"
11. Update the grade to something that will change the weight allowance (See `pkg/models/entitlements.go`)
12. Review
13. Additionally, you can set the `HasDependents` value on the order to true or false, then update the grade again to review how dependents affect the weight allowance.

## Screenshots
